src/code/relocation.c

```diff
@@ -25,71 +68,74 @@ void Overlay_Relocate(void* allocatedVRamAddress, OverlayRelocationSection* over
+            case R_MIPS_26 << REL_TYPE_SHIFT:
                 // Handles 26-bit address relocation, used for jumps and jals.
                 // Extract the address from the target field of the J-type MIPS instruction.
                 // Relocate the address and update the instruction.
 
-                unrelocatedAddress = PHYS_TO_K0((*relocDataP & 0x03FFFFFF) << 2);
-                relocOffset = unrelocatedAddress - (uintptr_t)vRamStart;
+                unrelocatedAddress = PHYS_TO_K0(MIPS_JUMP_TARGET(*relocDataP));
+                relocOffset = unrelocatedAddress - (uintptr_t)vramStart;
                 relocatedValue = (*relocDataP & 0xFC000000) | (((allocu32 + relocOffset) & 0x0FFFFFFF) >> 2);
-                relocatedAddress = PHYS_TO_K0((relocatedValue & 0x03FFFFFF) << 2);
+                relocatedAddress = PHYS_TO_K0(MIPS_JUMP_TARGET(relocatedValue));
                 *relocDataP = relocatedValue;
                 break;
 
-            case R_MIPS_HI16 << RELOC_TYPE_SHIFT:
+            case R_MIPS_HI16 << REL_TYPE_SHIFT:
                 // Handles relocation for a hi/lo pair, part 1.
                 // Store the reference to the LUI instruction (hi) using the `rt` register of the instruction.
                 // This will be updated later in the `R_MIPS_LO16` section.
 
-                luiRefs[(*relocDataP >> 0x10) & 0x1F] = relocDataP;
-                luiVals[(*relocDataP >> 0x10) & 0x1F] = *relocDataP;
+                luiRefs[MIPS_REG_RT(*relocDataP)] = relocDataP;
+                luiVals[MIPS_REG_RT(*relocDataP)] = *relocDataP;
                 break;
 
-            case R_MIPS_LO16 << RELOC_TYPE_SHIFT:
+            case R_MIPS_LO16 << REL_TYPE_SHIFT:
                 // Handles relocation for a hi/lo pair, part 2.
                 // Grab the stored LUI (hi) from the `R_MIPS_HI16` section using the `rs` register of the instruction.
                 // The full address is calculated, relocated, and then used to update both the LUI and lo instructions.
                 // If the lo part is negative, add 1 to the LUI value.
                 // Note: The lo instruction is assumed to have a signed immediate.
 
-                luiInstRef = luiRefs[(*relocDataP >> 0x15) & 0x1F];
-                regValP = &luiVals[(*relocDataP >> 0x15) & 0x1F];
+                luiInstRef = luiRefs[MIPS_REG_RS(*relocDataP)];
+                regValP = &luiVals[MIPS_REG_RS(*relocDataP)];
 
                 // Check address is valid for relocation
                 if ((((*regValP << 0x10) + (s16)*relocDataP) & 0x0F000000) == 0) {
-                    relocOffset = ((*regValP << 0x10) + (s16)*relocDataP) - (uintptr_t)vRamStart;
-                    isLoNeg = (((relocOffset + allocu32) & 0x8000) ? 1 : 0);
+                    relocOffset = ((*regValP << 0x10) + (s16)*relocDataP) - (uintptr_t)vramStart;
+                    isLoNeg = (((relocOffset + allocu32) & 0x8000) ? 1 : 0); // adjust for signed immediate
```

Comment:
```suggestion
                    isLoNeg = ((relocOffset + allocu32) & 0x8000) ? 1 : 0; // adjust for signed immediate
```
Perhaps just a personal preference.

Reply:
This is also my preference, I just left it as it was before. Changed